### ISSUE TITLE
Usability: When assigning already used channels, swap with channels that was previously used on the output that grabs the already used channels

### ISSUE THREAD

**Issue by spotlesscoder** [mixxxdj/mixxx#13858]
### Feature Description

Scenario: I have assigned channel 1-2 to headphones
Now I want to assign channel 3-4 to headphones and 1-2 to main out
So I choose the device for main and channels 3-4 automatically are assigned to main
Now I change the channels for main to 1-2
Now both main and headphones share channel 1-2 which is I guess something you almost never want

So the headphone channels should have been set to 3-4 (what was prior on main) automatically

**Comment by Swiftb0y** [comment]
but what if you do want want both outputs on the same channel? Automatically swapping would make it impossible. Also how often do you swap channels that this feature would be a meaningful speedup for you?

**Comment by spotlesscoder** [comment]
Maybe we can make a button that can revert the automatic swap? Or ask the user in a dialog?

Unfortunately Mixxx forgets my settings almost every time I unplug and re-plug my controller so very often

**Comment by Swiftb0y** [comment]
yeah that is an issue. Though we added some features to make that better in the next versions.

**Comment by ronso0** [comment]
> but what if you do want want both outputs on the same channel?

Not possible anyway ; )
![image](https://github.com/user-attachments/assets/205c9bee-d19c-4cd3-b9f8-68e77bd1fb04)

**Comment by Swiftb0y** [comment]
Not if you ask Mixxx, but technically you can when rerouting using an external tool (when going through pipewire for example).

**Comment by ronso0** [comment]
Auto-select was added in #11859
I think auto-swap can be implemented in [DlgPrefSound::deviceChannelsChanged()](https://github.com/mixxxdj/mixxx/blob/cf9c4fe33a0f14b2218579138a392bc9aa5f9d25/src/preferences/dialog/dlgprefsound.cpp#L856), probably with a helper like [DlgPrefSoundItem::selectFirstUnusedChannelIndex](https://github.com/mixxxdj/mixxx/blob/cf9c4fe33a0f14b2218579138a392bc9aa5f9d25/src/preferences/dialog/dlgprefsounditem.cpp#L136).
